technical-specs/001-Identity-MVP-Spec/001-Identity-Spec.md

```diff
@@ -0,0 +1,125 @@
+
+# App Center Identity Spec 
+
+This spec details the App Center Identity service that we will launch a public preview MVP.
+
+## Overview 
+
+### TL;DR 
+The goal is to deliver an identity management service in App Center that allows app developers to: 
+
+1. Create and manage an Azure AD B2C resource/tenant in App Center 
+2. Configure the Azure AD B2C resource/tenant for the following identity approaches in order of priority: Google, Facebook, Email/Password, Microsoft 
+3. View and search for user identities, and the ability to edit in Azure 
+4. Use the provided identity to integrate with other services in App Center: 
+   - Crashes and analytics events (i.e. Users of the app will be tagged to their crash/analytics data) 
+   - Push notifications (i.e. App developers can send push notifications to specific users)
+   - Data Storage/GDPR 
+
+## User scenarios 
+
+### Overview 
+
+We are prioritizing the following platforms and providers: 
+- Platforms: Android, iOS, Xamarin, React Native
+- Provider configuration in App Center Portal (in order of priority):  
+  - Google,  
+  - Facebook, 
+  - Email/password,  
+  - Microsoft
+
+- Additional providers can be configured through the B2C tenant directly
+
+Additional platforms will be targeted in the future:
+- Windows and PWA/SPA Support
+
+For all the use cases below, app developers need to have an active Azure subscription.
+
+### User scenarios
+
+#### 1. As an app developer, I can provision an Identity service (i.e. Azure AD B2C) for my app.
+
+User experience: 
+
+- Lupita is an app developer who wants to connect her existing Azure subscription to the App Center Identity service. 
+- She goes into the App Center Identity service, and she sees a button to “Connect to Azure subscription”. 
+- If: 
+  - She already has one or more Azure subscription(s) connected,  
+    - A panel pops up and she selects which the Azure subscription to connect her app to 
+  - Else, she is directed a panel “Add an Azure subscription”, she chooses her subscription and clicks “Connect” 
+    - If she does not have an Azure Subscription, she is prompted to “Create a subscription” and is redirected to the Azure portal to create a subscription 
+- A panel appears for her to input the domain name and choose the country/region to provision the AD B2C tenant/resource. Then, she clicks “Create”. 
```

Comment:
That's a very good point @burkeholland!

Our goal is to make it as simple as possible, and we explain this in our UI when the app developers sets up the tenant and the domain: 

![image](https://user-images.githubusercontent.com/33160470/52602783-cb838e00-2e18-11e9-978b-53afd7b9dbad.png)

Under "domain", there will be a tooltip explaining what it is:

![image](https://user-images.githubusercontent.com/33160470/52602806-ed7d1080-2e18-11e9-91b5-11f9d0be8c02.png)

Is this clear to you? Let us know how we can phrase it better!

Reply:
Are there cases a developer will need to go to the "full" b2c experience? If so, it may cause later friction if a developer will need to grok these concepts.

Reply:
I like the UI a lot, but It seems like we are still exposing people to our proprietary vocabulary. If they don't know what any of this is, they need to stop and do some research. Here are some questions that immediately pop into my head...

If I want username and password authentication, do I need an "AD B2C"? Why do I need an AD B2C and a tenant? Are those separate things? How much does one of these cost? Should I use one I already have or is it better to create a new one?

What if we hid all of that behind something like the following...

"Create a new identity provider"
and then maybe a collapsed details section that explains more details about
what that actually means (AD B2C).

Then for the domain screen, prefill the domain and make it a non-dominant
user entry field that people can edit if they like, but otherwise it's not
a prominent part of the UI.

What do you think?

Reply:
Yes, for greater customization of their settings, app developers are directed to the right page in the Azure AD B2C portal with an "Edit in Azure" button on each tab. The definitions used in the App Center portal (i.e. tooltips, banners) and documentation will be based on Azure AD B2C definitions. Great point!

![image](https://user-images.githubusercontent.com/33160470/52603780-1dc6ae00-2e1d-11e9-823d-86318b711e62.png)

Reply:
I think these AD details and terms are all friction points. It reduces the likelihood that a developer unfamiliar with Azure will get through the process of creating apps and using App Center during evaluation. My feeling on this is that we should only surface what the developer absolutely has to see to continue. What is the minimal amount of information, inputs and concepts that we can show to get them to success?

Reply:
Got it! I've set up a time with you today to discuss the user experience over a call. Thank you again for being active and giving us great feedback!